skyportal/models/phot_stat.py

```diff
@@ -566,42 +574,71 @@ def full_update(self, phot_list):
 
         Parameters
         ----------
-        phot_list: 1D array-like of skyportal.models.Photometry
+        phot_list: 1D array-like of skyportal.models.Photometry or dicts
             List of photometry points associated with this object.
 
         """
-        filters = np.array([phot.filter for phot in phot_list])
-        mjds = np.array([phot.mjd for phot in phot_list])
-        mags = np.array([phot.mag for phot in phot_list])
-        dets = np.array(
-            [
-                phot.flux
-                and phot.fluxerr
-                and phot.flux / phot.fluxerr > PHOT_DETECTION_THRESHOLD
-                for phot in phot_list
-            ],
-            dtype=bool,
-        )
-        lims = np.nan * np.ones(len(dets))
-
-        for i, phot in enumerate(phot_list):
+        if len(phot_list) == 0:
+            # use initialization to set None/{} to all values
+            self.__init__(self.obj_id)
+
+            # make sure to update the last update time
+            self.last_update = datetime.utcnow()
+            self.last_full_update = datetime.utcnow()
+            return
+
+        filters = []
+        mjds = []
+        mags = []
+        dets = []
+        lims = []
+        for phot in phot_list:
             if isinstance(phot, Photometry):
                 phot = phot.__dict__
-            if not dets[i]:  # get limiting magnitude for non-detection
+            elif not isinstance(phot, dict):
+                raise TypeError('phot must be a dict or Photometry object')
+
+            filters.append(phot['filter'])
+            mjds.append(phot['mjd'])
+
+            if 'mag' in phot:
+                mags.append(phot['mag'])
+            elif phot['flux'] is not None and phot['flux'] > 0:
+                mags.append(-2.5 * np.log10(phot['flux']) + PHOT_ZP)
+            else:
+                mags.append(np.nan)
+
+            is_detected = (
```

Comment:
Can't the presence of magnitude also indicate detection?

Reply:
I think you can have a 2 sigma detection with magnitude that doesn't count as a "detection".

Reply:
@guynir42 reasonable! Ok so to me should be ok